packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -166,10 +201,23 @@ export function initInput(
   checked: ?boolean,
   defaultChecked: ?boolean,
   type: ?string,
+  name: ?string,
   isHydrating: boolean,
 ) {
   const node: HTMLInputElement = (element: any);
 
+  if (
+    type != null &&
+    typeof type !== 'function' &&
+    typeof type !== 'symbol' &&
+    typeof type !== 'boolean'
+  ) {
+    if (__DEV__) {
+      checkAttributeStringCoercion(type, 'type');
+    }
+    node.type = type;
```

Comment:
This is always setting the type instead of only if it has changed. Not sure if that would matter like resetting dirty flags.